README.md

```diff
@@ -251,27 +251,70 @@ Every IPFS instance also exposes the libp2p API at `ipfs.libp2p`. The formal int
 
 ### [Runtime Support](https://github.com/ipfs/js-ipfs/issues/536)
 
+### Code Architecture and folder Structure
+
+![](/img/overview.png)
+
+##### Source code
+
+```Bash
+> tree src -L 2
+src                 # Main source code folder
+├── cli             # Implementation of the IPFS CLI
+│   └── ...
+├── http-api        # The HTTP-API implementation of IPFS as defined by http-api-spec
+├── core            # IPFS implementation, the core (what gets loaded in browser)
```

Comment:
Almost entirely agree with the caveats of:

- This forces us to run the 5 sets of tests every time
- It gets a way to people install js-ipfs as they are used to do it with go-ipfs (install one thing, get it all)